lib/process-services-cloud/src/lib/process-cloud/process-filters-cloud/process-filters-cloud.component.ts

```diff
@@ -0,0 +1,162 @@
+})
+export class ProcessFiltersCloudComponent implements OnChanges {
+
+    /** (required) The application name */
+    @Input()
+    appName: string;
+
+    /** (optional) The filter to be selected by default */
+    @Input()
+    filterParam: ProcessInstanceFilterRepresentationModel;
+
+    /** (optional) The flag hides/shows icon against each filter */
+    @Input()
+    showIcons: boolean = false;
+
+    /** Emitted when a filter is selected/clicked */
+    @Output()
+    filterClick: EventEmitter<ProcessInstanceFilterRepresentationModel> = new EventEmitter<ProcessInstanceFilterRepresentationModel>();
+
+    /** Emitted when filters are loaded successfully */
+    @Output()
+    success: EventEmitter<any> = new EventEmitter<any>();
+
+    /** Emitted when any error occurs while loading the filters */
+    @Output()
+    error: EventEmitter<any> = new EventEmitter<any>();
+
+    filters$: Observable<ProcessInstanceFilterRepresentationModel[]>;
+
+    currentFilter: ProcessInstanceFilterRepresentationModel;
+
+    filters: ProcessInstanceFilterRepresentationModel [] = [];
+
+    constructor(private processFilterCloudService: ProcessFilterCloudService) {
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        const appName = changes['appName'];
+        const filter = changes['filterParam'];
+        if (appName && appName.currentValue) {
+            this.getFilters(appName.currentValue);
+        } else if (filter && filter.currentValue !== filter.previousValue) {
+            this.selectFilter(filter.currentValue);
+        }
+    }
+
+    /**
+     * Return the filter list filtered by appName
+     */
+    getFilters(appName: string) {
+        this.filters$ = this.processFilterCloudService.getProcessInstanceFilters(appName);
+
+        this.filters$.subscribe(
+            (res: ProcessInstanceFilterRepresentationModel[]) => {
+                if (res.length === 0) {
+                    this.createFilters(appName);
+                } else {
+                    this.resetFilter();
+                    this.filters = res;
+                }
+                this.selectFilterAndEmit(this.filterParam);
+                this.success.emit(res);
+            },
+            (err: any) => {
+                this.error.emit(err);
+            }
+        );
+    }
+
+    /**
+     * Create default filters by appName
+     */
+    createFilters(appName?: string) {
+        this.filters$ =  this.processFilterCloudService.createDefaultFilters(appName);
+
+        this.filters$.subscribe(
+            (resDefault: ProcessInstanceFilterRepresentationModel[]) => {
+                this.resetFilter();
+                this.filters = resDefault;
+            },
+            (errDefault: any) => {
+                this.error.emit(errDefault);
+            }
+        );
+    }
+
+    /**
+     * Pass the selected filter as next
+     */
+    public selectFilter(newFilter: ProcessInstanceFilterRepresentationModel) {
```

Comment:
This method should be able to filter even for `id` and `index` not just for name